src/topology/config/watcher.rs

```diff
@@ -0,0 +1,125 @@
+use super::Config;
+use notify::{watcher, DebouncedEvent, RecommendedWatcher, RecursiveMode, Watcher};
+use std::{
+    path::{Path, PathBuf},
+    sync::mpsc::{channel, Receiver},
+    thread,
+    time::Duration,
+};
+
+/// Per notify own documentation, it's advised to have delay of more than 30 sec,
+/// so to avoid receiving repetitions of previous events on macOS.
+///
+/// But, config and topology reload logic can handle:
+///  - Invalid config, caused either by user or by data race.
+///  - Frequent changes, caused by user/editor modifying/saving file in small chunks.
+/// so we can use smaller, more responsive delay.
+pub const CONFIG_WATCH_DELAY: std::time::Duration = std::time::Duration::from_secs(1);
+
+const RETRY_TIMEOUT: std::time::Duration = std::time::Duration::from_secs(10);
+
+/// On unix triggers SIGHUP when file on config_path changes.
+/// Accumulates file changes until no change for given duration has occured.
+/// Has best effort guarante of detecting all file changes from the layend of
+/// this function until the main thread stops.
+///
+/// Doesn't do anything on Windows.
```

Comment:
On windows I think it may be best to error out if they have this setting set and to say something along the lines that this isn't currently supported on windows and the gh issue for it.

Reply:
I agree, just to report the error, and continue with running.